File: VB/PreviewSortSample/OrderDetailsReport.vb
Imports System
Imports DevExpress.XtraReports.UI

' ...
Namespace PreviewSortSample

    Public Partial Class OrderDetailsReport
        Inherits DevExpress.XtraReports.UI.XtraReport

        Public Sub New()
            Me.InitializeComponent()
        End Sub

        Public Sub SetOrderID(ByVal orderID As Integer)
            Me.nwindDataSet1.Clear()
            Me.orderDetailsTableAdapter.Fill(Me.nwindDataSet1.OrderDetails, orderID)
            Me.lbOrderID.Text = orderID.ToString()
        End Sub

        Public Sub Sort(ByVal fieldName As String)
            ' Disable sorting.
            Me.Detail.SortFields.Clear()
            ' Create a new sorting criterion.
            Dim grField As DevExpress.XtraReports.UI.GroupField = New DevExpress.XtraReports.UI.GroupField With {.FieldName = fieldName, .SortOrder = DevExpress.XtraReports.UI.XRColumnSortOrder.Ascending}
            ' Enable sorting.
            Me.Detail.SortFields.Add(grField)
            ' Re-create the report document.
            Me.CreateDocument(True)
        End Sub
    End Class
End Namespace

File: CS/PreviewSortSample/OrderDetailsReport.cs
using System;
using DevExpress.XtraReports.UI;
// ...

namespace PreviewSortSample {
    public partial class OrderDetailsReport : XtraReport {
        public OrderDetailsReport() {
            InitializeComponent();
        }

        public void SetOrderID(int orderID) {
            nwindDataSet1.Clear();
            orderDetailsTableAdapter.Fill(nwindDataSet1.OrderDetails, orderID);
            lbOrderID.Text = orderID.ToString();
        }

        public void Sort(string fieldName) {
            // Disable sorting.
            Detail.SortFields.Clear();

            // Create a new sorting criterion.
            GroupField grField = new GroupField { FieldName = fieldName, SortOrder = XRColumnSortOrder.Ascending };

            // Enable sorting.
            Detail.SortFields.Add(grField);

            // Re-create the report document.
            CreateDocument(true);
        }
    }
}
